metacat-thrift/src/main/java/com/netflix/metacat/thrift/CatalogThriftHiveMetastore.java

```diff
@@ -295,7 +312,15 @@ public AddPartitionsResult add_partitions_req(final AddPartitionsRequest request
      */
     @Override
     public void alter_database(final String dbname, final Database db) throws TException {
-        throw unimplemented("alter_database", new Object[]{dbname, db});
+        requestWrapper("update_database", new Object[]{db}, () -> {
+            if (dbname == null || db == null) {
+                throw new InvalidInputException("Invalid database request");
+            }
+            final DatabaseCreateRequestDto dto = new DatabaseCreateRequestDto();
+            dto.setMetadata(db.getParameters());
+            v1.updateDatabase(catalogName, normalizeIdentifier(dbname), dto);
+            return null;
```

Comment:
For this normalizeIdentifier() function, what if the dbname is an empty string? I guess it will pass into the metacat controller and finally we will catch it once reach the service layer. Wondering if we should catch it here?  
```
if (s == null) {
            return null;
        } else {
            return s.trim().toLowerCase(Locale.ENGLISH);
        }
```

Reply:
This method does not validate the name but just changes the case. The validation happens else where. Probably we could move up the validation in here. This will be a separate change.